src/ck3/data/religions.rs

```diff
@@ -69,7 +71,44 @@ impl DbKind for Religion {
         vd.field_item("family", Item::ReligionFamily);
 
         vd.req_field("doctrine");
-        vd.multi_field_item("doctrine", Item::Doctrine);
+        let mut categories: FnvHashMap<&str, Vec<Token>> = FnvHashMap::default();
+        vd.multi_field_validated_value("doctrine", |_, mut vd| {
+            vd.item(Item::Doctrine);
+            if let Some(category) = data.doctrines.category(vd.value().as_str()) {
+                let doctrine = vd.value();
+                if let Some(seen) = categories.get_mut(category.as_str()) {
+                    let picks = data.doctrines.number_of_picks(category.as_str());
+                    #[allow(clippy::cast_possible_wrap)]
+                    if picks == 1 {
+                        // SAFETY: we never push empty vecs into this hash
+                        let other_doctrine = &seen[0];
+                        if doctrine == other_doctrine {
+                            let msg = format!("duplicate doctrine {doctrine}");
+                            warn(ErrorKey::DuplicateField)
+                                .msg(msg)
+                                .loc_msg(doctrine, "doctrine")
+                                .loc(other_doctrine, "earlier doctrine")
+                                .push();
+                        } else {
+                            let msg =
+                                format!("{doctrine} and {other_doctrine} are both from {category}");
+                            err(ErrorKey::Conflict)
+                                .msg(msg)
+                                .loc_msg(doctrine, "doctrine")
+                                .loc(other_doctrine, "earlier doctrine")
+                                .push();
+                        }
+                    } else if picks == (seen.len() as i64) {
+                        let msg =
+                            format!("religion has more than {picks} doctrines from {category}");
```

Comment:
perhaps we can link it to the `number_of_picks = X` line by storing a loc alongside the actual value

Reply:
Yeah though I think I'll do it so that picks is simply preserved as a Token and then decoded where it's used.
That's the intended use of Token, at leas